test/test_belief_propagation.jl

```diff
@@ -95,4 +106,44 @@ using SplitApplyCombine
   eigs = eigvals(rdm)
   @test all(>=(0), real(eigs)) && all(==(0), imag(eigs))
   @test size(rdm) == (2^length(vs), 2^length(vs))
+
+  #TEST MORE ADVANCED BP WITH ITENSORNETWORK MESSAGE TENSORS. IN THIS CASE IT SHOULD BE LIKE BOUNDARY MPS
+  g_dims = (5, 4)
+  g = named_grid(g_dims)
+  s = siteinds("S=1/2", g)
+  χ = 2
+  ψ = randomITensorNetwork(s; link_space=χ)
+  maxdim = 16
+  v = (2, 2)
+
+  ψψ = flatten_networks(ψ, dag(ψ); combine_linkinds=false, map_bra_linkinds=prime)
+  Oψ = copy(ψ)
+  Oψ[v] = apply(op("Sz", s[v]), ψ[v])
+  ψOψ = flatten_networks(ψ, dag(Oψ); combine_linkinds=false, map_bra_linkinds=prime)
+
+  combiners = linkinds_combiners(ψψ)
+  ψψ = combine_linkinds(ψψ, combiners)
+  ψOψ = combine_linkinds(ψOψ, combiners)
+
+  nsites = 1
+  vertex_groups = nested_graph_leaf_vertices(partition(ψψ, group(v -> v[1], vertices(ψψ))))
+  mts = compute_message_tensors(
```

Comment:
Can we rename `compute_message_tensors(...)` to `belief_propagation(...)`? It looks like it is the same function except it picks some default for the initial message tensors, in general when that is the case (i.e. two functions are the same but one sets some default inputs) I would name them the same thing.

Reply:
Good idea.